scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/termination/handlers/TerminateIfTaskHandler.java

```diff
@@ -0,0 +1,190 @@
+package org.ow2.proactive.scheduler.job.termination.handlers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.ow2.proactive.scheduler.common.NotificationData;
+import org.ow2.proactive.scheduler.common.SchedulerEvent;
+import org.ow2.proactive.scheduler.common.task.TaskId;
+import org.ow2.proactive.scheduler.common.task.TaskStatus;
+import org.ow2.proactive.scheduler.common.task.flow.FlowAction;
+import org.ow2.proactive.scheduler.core.SchedulerStateUpdate;
+import org.ow2.proactive.scheduler.job.ChangedTasksInfo;
+import org.ow2.proactive.scheduler.job.InternalJob;
+import org.ow2.proactive.scheduler.job.JobInfoImpl;
+import org.ow2.proactive.scheduler.task.internal.InternalTask;
+
+public class TerminateIfTaskHandler {
+
+	public static final Logger logger = Logger.getLogger(TerminateIfTaskHandler.class);
+
+	private final InternalJob internalJob;
+
+	public TerminateIfTaskHandler(InternalJob internalJob) {
+		this.internalJob = internalJob;
+	}
+
+	public boolean terminateIfTask(FlowAction action, InternalTask initiator, ChangedTasksInfo changesInfo,
+			SchedulerStateUpdate frontend, InternalTask descriptor, TaskId taskId) {
+
+		InternalTask[] targets = searchIfElseJoinTasks(action, initiator);
+
+		// the targetIf from action.getTarget() is the selected branch;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)